modules/bastion/variables.tf

```
@@ -0,0 +1,126 @@
+}
+
+variable "associate_public_ip_address" {
+  type        = bool
+  default     = false
+  description = "Whether to associate public IP to the instance."
+}
+
+variable "ebs_block_device_volume_size" {
+  type        = number
+  default     = 0
+  description = "The volume size (in GiB) to provision for the EBS block device. Creation skipped if size is 0"
+}
+
+variable "ebs_delete_on_termination" {
+  type        = bool
+  default     = false
+  description = "Whether the EBS volume should be destroyed on instance termination"
+}
+
+variable "user_data" {
+  type        = list(string)
+  default     = []
+  description = "User data content"
+}
+
+variable "security_group_rules" {
+  type = list(any)
+  default = [
+    {
+      type        = "egress"
+      from_port   = 0
+      to_port     = 0
+      protocol    = -1
+      cidr_blocks = ["0.0.0.0/0"]
+    },
+    {
+      type        = "ingress"
+      protocol    = "tcp"
+      from_port   = 22
+      to_port     = 22
+      cidr_blocks = ["0.0.0.0/0"]
+    }
+  ]
+  description = <<-EOT
+    A list of maps of Security Group rules.
+    The values of map is fully complated with `aws_security_group_rule` resource.
+    To get more info see https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule .
+  EOT
+}
+
+variable "custom_bastion_hostname" {
+  type        = string
+  default     = null
+  description = "Hostname to assign with bastion instance"
+}
+
+variable "vanity_domain" {
+  type        = string
+  default     = null
+  description = "Vanity domain"
+}
+
+variable "ssh_key_path" {
+  type        = string
+  default     = "./secrets"
+  description = "Save location for ssh public keys generated by the module"
+}
+
+variable "generate_ssh_key" {
+  type        = bool
+  default     = true
+  description = "Whether or not to generate an SSH key"
+}
+
+# AWS KMS alias used for encryption/decryption of SSM secure strings
+variable "kms_alias_name_ssm" {
+  type        = string
+  default     = "alias/aws/ssm"
+  description = "KMS alias name for SSM"
+}
+
+variable "container_enabled" {
+  type        = bool
+  default     = false
+  description = "Enable or disable container functionality."
+}
+
+variable "container_command" {
+  type        = string
+  default     = "bash"
+  description = "The container command passed in after `docker run --rm -it <image> bash -c`."
+}
+
+variable "image_repository" {
+  type        = string
+  default     = ""
+  description = "The image repository to use in `container.sh`. This is required if `container_enabled` is `true`."
+}
+
+variable "image_container" {
+  type        = string
+  default     = ""
+  description = "The image container to use in `container.sh`. This is required if `container_enabled` is `true`."
+}
+
+variable "ssh_pub_keys" {
+  type        = bool
+  default     = false
+  description = "Enable ssh pub keys from chamber."
+}
```

It'll be nice if this variable existed. Not that it's a good idea to disable it, but I'd say it's an important configuration that shouldn't be abstracted at the component level.

```suggestion
}

variable "ssm_enabled" {
  description = "Enable SSM Agent on Host."
  type        = bool
  default     = true
}
```